cmd/infra/powervs/create.go

```diff
@@ -0,0 +1,1170 @@
+package powervs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/go-logr/logr"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/networking-go-sdk/directlinkv1"
+	"github.com/IBM/networking-go-sdk/zonesv1"
+	"github.com/IBM/platform-services-go-sdk/globalcatalogv1"
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/platform-services-go-sdk/resourcemanagerv2"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+
+	hypershiftLog "github.com/openshift/hypershift/cmd/log"
+)
+
+var cloudApiKey = os.Getenv("IBMCLOUD_API_KEY")
+
+const (
+	// Resource name suffix for creation
+	cloudInstanceNameSuffix = "nodepool"
+	vpcNameSuffix           = "vpc"
+	vpcSubnetNameSuffix     = "vpc-subnet"
+	cloudConnNameSuffix     = "cc"
+
+	// Default cloud connection speed
+	defaultCloudConnSpeed = 5000
+
+	// CIS service name
+	cisService = "internet-svcs"
+
+	// PowerVS service and plan name
+	powerVSService     = "power-iaas"
+	powerVSServicePlan = "power-virtual-server-group"
+
+	// Resource desired states
+	vpcAvailableState               = "available"
+	cloudInstanceActiveState        = "active"
+	dhcpServiceActiveState          = "ACTIVE"
+	cloudConnectionEstablishedState = "established"
+
+	// Resource undesired state
+	dhcpServiceErrorState = "ERROR"
+
+	// Time duration for monitoring the resource readiness
+	pollingInterval                  = time.Second * 5
+	vpcCreationTimeout               = time.Minute * 5
+	cloudInstanceCreationTimeout     = time.Minute * 5
+	cloudConnEstablishedStateTimeout = time.Minute * 30
+	dhcpServerCreationTimeout        = time.Minute * 30
+	cloudConnUpdateTimeout           = time.Minute * 10
+
+	// Service Name
+	powerVsService  = "powervs"
+	vpcService      = "vpc"
+	platformService = "platform"
+)
+
+var customEpEnvNameMapping = map[string]string{
```

Comment:
Are this env variables documented somewhere? how do I know what to do with them?

Reply:
Added this to the How-To Doc